app/battle_preparation.py

```diff
@@ -0,0 +1,11 @@
+from app.knight_class import Knight
+
+
+def battle_preparation(knight: Knight, knight_dict: dict) -> object:
+    knight.protection += sum([armor["protection"] for armor in knight_dict["armour"]])
```

Comment:
you can use `knight.protection += sum(armor["protection"] for armor in knight_dict["armour"]) ` . because it is useless to create list here